app/server/routes/api.ts

```diff
@@ -5,14 +5,14 @@ import {
   MDA_TEST_USER_HEADER,
   MembersDataApiItem
 } from "../../shared/productResponse";
+import { apiGatewayHandler } from "../apiGatewayHandler";
```

Comment:
IMO, handler is a meaningless name which does not convey intention. How about `discoverAuthorizedResource` or something which conveys host+key intention.